contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -906,6 +907,26 @@ class tsqlCommonMutator : public TSqlParserBaseListener
 	void exitFunction_call(TSqlParser::Function_callContext *ctx) override
 	{
 		handleGeospatialFunctionsInFunctionCall(ctx);
+
+		if (ctx->func_proc_name_server_database_schema())
+		{
+			auto fpnsds = ctx->func_proc_name_server_database_schema();
+
+			if (fpnsds->DOT().empty() && fpnsds->id().back()->keyword()) /* built-in functions */
+			{
+				auto id = fpnsds->id().back();
+
+				if (id->keyword()->TRIM())
+				{
+					rewritten_query_fragment.emplace(std::make_pair(id->keyword()->TRIM()->getSymbol()->getStartIndex(), std::make_pair(::getFullText(id->keyword()->TRIM()), "sys.trim")));
+				}
+			}
+		}
+	}
+
+	void exitTRIM(TSqlParser::TRIMContext *ctx) override
```

Comment:
Why do we need this if we are already handling it in `exitFunction_call`?

Reply:
Function call `TRIM(expression FROM expression)` is getting called in `function_call->built_in_functions->bif_other`, so this above handling inside exitTRIM is for that. Whereas Function call `TRIM(expression)` is getting called in `function_call->func_proc_name_server_database_schema` hence handling for that is added in exitFunctionCall as well.